### PR TITLE
275 announcement feature adding support for guild member property and handling discord bot permissions

### PR DESCRIPTION
**Pull request by Behzad-rabiei**

### CODE REVIEW

File: src/config/index.ts
Comment (scientiststwin) [inline src/config/index.ts:1]: @Behzad-rabiei We should not allow `REDIS_PASSWORD` to be empty?  🤔

File: src/config/index.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ const envVarsSchema = Joi.object()
         SESSION_SECRET: Joi.string().required().description('Session secret'),
         REDIS_HOST: Joi.string().required().description('Redis host'),
         REDIS_PORT: Joi.string().required().description('Redis port'),
-        REDIS_PASSWORD: Joi.string().required().description('Redis password'),
+        REDIS_PASSWORD: Joi.string().required().description('Reids password').allow(''),
```
Comment (scientiststwin) [inline src/config/index.ts:43]: @Behzad-rabiei We should not allow `REDIS_PASSWORD` to be empty?  🤔

Comment (Behzad-rabiei) [reply src/config/index.ts:43]: @scientiststwin Redis server can be use without password.

Comment (Behzad-rabiei) [reply src/config/index.ts:43]: Redis enters a special mode called protected mode when it is executed with the default configuration (binding all the interfaces) and without any password in order to access it.

**Pull request merged**